### PR TITLE
Trying to reduce mac installer size

### PR DESCRIPTION
**Pull request by wpotrzebowski**
It seems to be reducing the Mac installer size by ~200MB 

## Description

It enables UPX compression for Linux and OSX installers. 

## Review Checklist (please remove items if they don't apply):

- [ ] Code has been reviewed
- [ ] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [ ] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [ ] User documentation is available and complete (if required)
- [x] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

**Review by rozyczko — COMMENTED** [review]
does indeed limit the size of the macos image by ~200MB.

**Review by rozyczko — COMMENTED** [review]
Works on windows, needs checking the macos installer

**Comment by andyfaff** [comment]
When the numpy and scipy wheels use accelerate the size will reduce even further.

**Comment by wpotrzebowski** [comment]
Thanks for the hint @andyfaff! Do you have an example of how to install accelerate wheels with pip? Does one also have to make changes in pyinstaller (spec file)?

**Comment by andyfaff** [comment]
Wheels that use accelerate aren't available yet.  Guessing either the next release, or the one after, for both packages.

**Comment by butlerpd** [comment]
Should this also be merged into main? Otherwise this looks ready to merge? Assuming it has been tested on Mac OS?

**Comment by butlerpd** [comment]
May need testing on MacOS intel architecture? @pkienzle will check.

**Comment by pkienzle** [comment]
Failed. Docs not available in `SasView6.app/Contents/MacOS/docs`, though I do see them in `SasView6.app/Contents/Resources/doc/{source,build}`

```
$ ~/Desktop/SasView6.app/Contents/MacOS/sasview   
Attribute Qt::AA_ShareOpenGLContexts must be set before QCoreApplication is created.
14:42:35 - sas.qtgui.MainWindow.MainWindow - ERROR - Application failed with: [Errno 2] No such file or directory: '…/SasView6.app/Contents/MacOS/docs/sphinx-docs/source-temp'
14:42:35 - sas.qtgui.MainWindow.GuiManager - ERROR - Traceback (most recent call last):
  File "sasview.py", line 15, in <module>
  File "sas/cli.py", line 161, in main
  File "sas/qtgui/MainWindow/MainWindow.py", line 106, in run_sasview
  File "sas/qtgui/MainWindow/MainWindow.py", line 47, in __init__
  File "sas/qtgui/MainWindow/MainWindow.py", line 44, in __init__
  File "sas/qtgui/MainWindow/GuiManager.py", line 103, in __init__
  File "sas/sascalc/doc_regen/makedocumentation.py", line 60, in create_user_files_if_needed
  File "shutil.py", line 559, in copytree
FileNotFoundError: [Errno 2] No such file or directory: '…/SasView6.app/Contents/MacOS/docs/sphinx-docs/source-temp'

[76208] Failed to execute script 'sasview' due to unhandled exception: [Errno 2] No such file or directory: '…/SasView6.app/Contents/MacOS/docs/sphinx-docs/source-temp'
[76208] Traceback:
Traceback (most recent call last):
  File "sasview.py", line 15, in <module>
  File "sas/cli.py", line 161, in main
  File "sas/qtgui/MainWindow/MainWindow.py", line 106, in run_sasview
  File "sas/qtgui/MainWindow/MainWindow.py", line 47, in __init__
  File "sas/qtgui/MainWindow/MainWindow.py", line 44, in __init__
  File "sas/qtgui/MainWindow/GuiManager.py", line 103, in __init__
  File "sas/sascalc/doc_regen/makedocumentation.py", line 60, in create_user_files_if_needed
  File "shutil.py", line 559, in copytree
FileNotFoundError: [Errno 2] No such file or directory: '…/SasView6.app/Contents/MacOS/docs/sphinx-docs/source-temp'
```

**Comment by wpotrzebowski** [comment]
Hmm, interesting. It seems to work for me either from double clicking on installer or from command line.

**Comment by krzywon** [comment]
> Hmm, interesting. It seems to work for me either from double clicking on installer or from command line.

The error @pkienzle is seeing is related to the doc generation. @wpotrzebowski  - Try deleting your `~/.sasview/doc/` directory and then running the app. I expect you see the same error. This is being addressed in #2795, but there are other Mac related issues that I am still trying to solve.

**Comment by wpotrzebowski** [comment]
You are right @krzywon. Same error when I removed `~/.sasview/doc/`

**Comment by krzywon** [comment]
As an added benefit, moving MacOS to pyinstaller v6 fixes #2813

**Comment by butlerpd** [comment]
@krzywon to check with @pkienzle.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -195,10 +195,15 @@ jobs:
...
       - name: Install utilities to build installer
-        if: ${{ matrix.installer }}
+        if: ${{ matrix.installer && startsWith(matrix.os, 'windows') }}
         run: |
           python -m pip install pyinstaller==5.13.2
 
+      - name: Install utilities to build installer
+        if: ${{ matrix.installer && !startsWith(matrix.os, 'windows') }}
+        run: |
+          python -m pip install pyinstaller==6.3.0
```
Comment (pkienzle) [inline .github/workflows/ci.yml:205]: Remove previous block, which installs an older version of pyinstaller

**Review by pkienzle — COMMENTED** [review]
Installs and runs on Intel Mac (OSX 13.6.5)

I was able to edit the pringle model docs in place, changing `sinc^2` to `\,\mathrm{sinc}^2`.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (krzywon) [reply .github/workflows/ci.yml:205]: Pyinstaller v6 isn't working for our Windows build, yet. The previous block is to allow Windows to use pyinstaller v5.x, conditional included.

**Review by butlerpd — APPROVED** [review]
Speaking with @pkienzle he misread the code and agrees with @krzywon now. He is leaving so am approving in his stead.

**Pull request merged**